### PR TITLE
Add EIS burn severity data

### LINKED ISSUE #206: Publish EIS Annual Fire-Hydro COGs
# Background notes
There are 5 annual Fire-Hydro COGs currently available in `veda-data-store-staging`. I did a quick spotcheck and the COGs appear valid and ready to ingest. We need to create a STAC collection, ingest these 5 COGs, and configure a dataset in delta-config.

**Data prefix:** `EIS/COG/Fire-Hydro`
**Example key:** `3://veda-data-store-staging/EIS/COG/Fire-Hydro/mtbs_CONUS_2016.cog.tif`
[cog/viewer preview](https://staging-raster.delta-backend.com/cog/viewer?url=s3://veda-data-store-staging/EIS/COG/Fire-Hydro/mtbs_CONUS_2016.cog.tif)

# Research + prepare processing for the dataset: Identify the dataset and what the processing needs are

1. Identify dataset and where it will be accessed from (HTTP from DAAC vs S3, for example). Check it's a good source with science team. Ask about specific variables and required spatial and temporal extent. Note many datasets will require back processing (e.g. generating cloud-optimized data for historical data).

    [Future: 2. If the dataset is ongoing (i.e. new files are continuously added and should be included in the dashboard), design and construct the scheduling + forward-processing workflow.]

2. If necessary, create COG or any other conversion / processing code and verify the COG output with a data product expert (for example, someone at the DAAC which hosts the native format) by sharing in a visual interface.

3. Identify the point of contact and ensure someone is providing them updates!

# Design the metadata and publish to the Dev API

1. If not already familiar with these conventions for generating STAC collection and item metadata:
       - Collections: https://github.com/NASA-IMPACT/delta-backend/issues/29 and STAC version 1.0 specification for collections
       - Items: https://github.com/NASA-IMPACT/delta-backend/issues/28 and STAC version 1.0 specification for items
       - NOTE: The delta-backend instructions are specific to datasets for the climate dashboard, however not all datasets are going to be a part of the visual layers for the dashboard so you can ignore the instructions that are specific to "dashboard" extension, "item_assets" in the collection and "cog_default" asset type in the item.

      A collection will need the following fields, some of which may be self-evident through the filename or an about page for the product, however there are many cases in which we may need to reach out to product owners to define the right values for these fields:
      
      - temporal interval
      - license
      - id
      - title
      - description
      - whether it is periodic or not on the dashboard
      - the dashboard time density
  
4. Review and follow https://github.com/NASA-IMPACT/cloud-optimized-data-pipelines/blob/main/OPERATING.md

## Publish to the Staging API

Once the PR is approved, we can merge and publish those datasets to the Staging API

---

## AC
- [ ] STAC collection and item records published to staging-stac.delta-backend.com
- [ ] PR submitted to delta-config to add this collection to the EIS thematic area as a new dataset (or delta-config issue generated and help requested)

### PR DESCRIPTION
**Pull request by ividito**
Closes #206 

Adds one collection with 5 static assets (annual from 2016-2020)

### CODE REVIEW

File: data/step_function_inputs/mtbs-burn-severity.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "collection": "mtbs-burn-severity",
+    "discovery": "s3",
+    "prefix": "EIS/COG/Fire-Hydro/",
+    "bucket": "veda-data-store-staging",
+    "filename_regex": "^(.*).tif$",
+    "datetime_range": "year",
+    "cogify": "true"
```
Comment (moradology) [inline data/step_function_inputs/mtbs-burn-severity.json:8]: Shouldn't be necessary here, I think. These are already cogs

File: data/collections/mtbs-burn-severity.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    "24.0478678762251",
+                    "-71.50752568733597",
+                    "50.55916724898132"
+                ]
+            ]
+        },
+        "temporal":{
+            "interval":[
+                [
+                    "2016-01-01",
```
[the remark this answers is not in the document]
Comment (anayeaye) [reply data/collections/mtbs-burn-severity.json:21]: yeah I think so: https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#temporal-extent-object
`[["2016-01-01T00:00:00Z", "2020-12-31T23:59:59Z"]]`

File: data/step_function_inputs/mtbs-burn-severity.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        "bucket": "veda-data-store-staging",
+        "filename_regex": "mtbs_CONUS_2016.cog.tif",
+        "start_datetime": "2016-01-01T00:00:00Z",
+        "end_datetime": "2016-12-31T23:59:59Z",
+    },
+    {    "collection": "mtbs-burn-severity",
+        "discovery": "s3",
+        "prefix": "EIS/COG/Fire-Hydro/",
+        "bucket": "veda-data-store-staging",
+        "filename_regex": "2017.cog.tif",
```
Comment (moradology) [inline data/step_function_inputs/mtbs-burn-severity.json:22]: Looks to me like this cog name might be a bit off. Compare `mtbs_CONUS_2016.cog.tif` vs `2017.cog.tif`

**Review by moradology — CHANGES_REQUESTED** [review]

**Comment by anayeaye** [comment]
@ividito the metadata ingested to the dev stack looks good to me but is there a newer version of the step function input json than what is checked in to this PR? It doesn't look like it would have captured the 2020 COG. Other than that this looks good. I'll 👍  after that update

**Comment by ividito** [comment]
> @ividito the metadata ingested to the dev stack looks good to me but is there a newer version of the step function input json than what is checked in to this PR? It doesn't look like it would have captured the 2020 COG. Other than that this looks good. I'll 👍 after that update

No newer version, but I think there might still be data from the initial ingestion with the broad regex instead of the entry-per-file. I've updated the PR with a fix.

**Comment by anayeaye** [comment]
> > @ividito the metadata ingested to the dev stack looks good to me but is there a newer version of the step function input json than what is checked in to this PR? It doesn't look like it would have captured the 2020 COG. Other than that this looks good. I'll 👍 after that update
> 
> No newer version, but I think there might still be data from the initial ingestion with the broad regex instead of the entry-per-file. I've updated the PR with a fix.

The updated version looks good to me

**Review by anayeaye — APPROVED** [review]

**Comment by ividito** [comment]
@gmarichalgxc @Catalina-Moller @mmaniceraGXC this dataset has been deployed, and the explore view of the new dataset needs to be tested. Deploy preview: https://deploy-preview-127--visex.netlify.app/eis/datasets/mtbs-burn-severity

File: data/collections/mtbs-burn-severity.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            "type": "image/tiff; application=geotiff; profile=cloud-optimized",
+            "roles": [
+                "data",
+                "layer"
+            ],
+            "title": "Default COG Layer",
+            "description": "Cloud optimized default layer to display on map"
+        }
+    }
+}
```
Comment (slesaad) [inline data/collections/mtbs-burn-severity.json:43]: newline

**Review by slesaad — APPROVED** [review]

**Pull request merged**